### PR TITLE
[ALLUXIO-1856] When authorization is not turned on, changed the default permission to full permission

### PR DESCRIPTION
**Pull request by yupeng9**
By default the security and authorization is not turned on. However, the permission on inodes by default was set to no permission. This is problematic when permission info is fetched by listStatus. The default permission should be full permission.

### CODE REVIEW

**Comment by jsimsa** [comment]
@apc999 PTAL

why are permissions problematic when security is disabled? when security it disabled, permissions should not have any effect

**Comment by apc999** [comment]
@jsimsa when security flags are turned off, there is no permission check on filesystem operations. However, the permission field is still in inode (just not used). when you create a new inode, you need to fill a default permission in that case.

**Comment by apc999** [comment]
@yupeng9 , could you fix the failing unit tests?

**Comment by apc999** [comment]
otherwise the PR looks good to me

**Comment by jsimsa** [comment]
why does the default value matters if the security is off?

**Comment by ifcharming** [comment]
@jsimsa Alluxio internally does not check any permission the security is disabled. However, if another framework try to access Alluxio as `AbstractFileSystem`, such as using `hadoop fs`, permission can be checked.

**Comment by ifcharming** [comment]
LGTM, thanks for the important fix!

Please fix the failing tests.

**Comment by ifcharming** [comment]
I think the most recent test failures are because:
"AuthenticatedClientUser.set(TEST_AUTHENTICATE_USER);” is missing in call() for ConcurrentFree and ConcurrentDelete.

**Comment by yupeng9** [comment]
@ifcharming thanks for the tip.

**Comment by yupeng9** [comment]
Jenkins, test this please

File: tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java [hunk 1]
```diff
@@ -758,6 +761,8 @@ public void ttlCreateFileTest() throws Exception {
     Assert.assertEquals(ttl, folderInfo.getTtl());
   }
 
+  @LocalAlluxioClusterResource.Config(
```
Comment (apc999) [inline tests/src/test/java/alluxio/master/file/FileSystemMasterIntegrationTest.java]: I think the previous convention is to have `@Test` first,
then `@LocalAlluxioClusterResource.Config`

File: core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java [hunk 2]
```diff
@@ -85,10 +85,14 @@ public FileSystemPermission getPermission() {
    * Applies umask.
    *
    * @param umask the umask to apply
+   * @param configuration the configuration
    * @return a new {@link PermissionStatus}
    * @see FileSystemPermission#applyUMask(FileSystemPermission)
    */
-  public PermissionStatus applyUMask(FileSystemPermission umask) {
+  public PermissionStatus applyUMask(FileSystemPermission umask, Configuration configuration) {
+    if (!SecurityUtils.isAuthorizationEnabled(configuration)) {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java]: I feel this check should not be here inside `applyUMask` but on the caller side.
The reason is that, by having this, when security is not turned on, `PermissionStatus.applyUMask` will become no-op, which makes less sense. However, I feel this `applyUMask` logic should still hold as it is a part of `PermissionStatus` logic, it should not depend on whether security is turned on or off.

File: core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java [hunk 3]
```diff
@@ -117,9 +121,9 @@ public static PermissionStatus getDirDefault() {
    * @throws java.io.IOException when getting login user fails
    */
   public static PermissionStatus get(Configuration conf, boolean remote) throws IOException {
-    if (!SecurityUtils.isAuthenticationEnabled(conf)) {
-      // no authentication
-      return new PermissionStatus("", "", FileSystemPermission.getNoneFsPermission());
+    if (!SecurityUtils.isSecurityEnabled(conf)) {
```
Comment (apc999) [inline core/common/src/main/java/alluxio/security/authorization/PermissionStatus.java]: is this change necessary?

**Comment by apc999** [comment]
@yupeng9  actually could you elaborate on the question raised by @jsimsa : why returning Null Permission in file info "is problematic when permission info is fetched by listStatus"?

**Comment by yupeng9** [comment]
@jsimsa @apc999 
We have other integrations such as `hadoop.fs` or fuse that will obtain the permission info from Alluxio, even if the security is not turned on. 

By default the file should be granted full permission when security is not turned on, otherwise applications may not be able to operate over Alluxio.

For example, hive requires the `/user/hive/warehouse` and `/tmp` to have write permission. Without this patch, hive cannot run on top of Alluxio.

**Comment by yupeng9** [comment]
Please review it at https://github.com/Alluxio/alluxio/pull/2994
